drivers/interrupt_controller/intc_plic.c

```diff
@@ -267,15 +311,27 @@ void riscv_plic_irq_disable(uint32_t irq)
 int riscv_plic_irq_is_enabled(uint32_t irq)
 {
 	const struct device *dev = get_plic_dev_from_irq(irq);
-	const struct plic_config *config = dev->config;
 	const uint32_t local_irq = irq_from_level_2(irq);
-	mem_addr_t en_addr = config->irq_en + local_irq_to_reg_offset(local_irq);
+	uint32_t bit_position = local_irq & PLIC_REG_MASK;
 	uint32_t en_value;
+	int is_enabled = IS_ENABLED(CONFIG_PLIC_IRQ_AFFINITY) ? 0 : 1;
+	uint32_t key = irq_lock();
+
+	for (uint32_t cpu_num = 0; cpu_num < arch_num_cpus(); cpu_num++) {
```

Comment:
My one concern here is that we've gone from `O(0)` to `O(n)`. Is there any way that we could pre-compute these values and store them in a LUT?

Reply:
I think the fact that `hartid != cpu_id` in RISC-V might make the build-time generation of LUT tricky, but might be possible. Runtime generated LUT should be easy.

But considering that we support up to 12 CPUs by default, and most in tree SoCs do not go over 8, are these APIs used frequently enough to make an impact to the performance?